MailCore/Cache/ContactManager/ContactManager+DB.swift

```diff
@@ -0,0 +1,72 @@
+/*
+ Infomaniak Mail - iOS App
+ Copyright (C) 2022 Infomaniak Network SA
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Foundation
+import RealmSwift
+
+public extension ContactManager {
+    func getRealm() -> Realm {
+        do {
+            return try Realm(configuration: realmConfiguration)
+        } catch {
+            // We can't recover from this error but at least we report it correctly on Sentry
+            Logging.reportRealmOpeningError(error, realmConfiguration: realmConfiguration)
+        }
+    }
+
+    /// Both *case* insensitive __and__ *diacritic* (accents) insensitive
+    static let searchContactInsensitivePredicate = "name contains[cd] %@ OR email contains[cd] %@"
+
+    func contacts(matching string: String) -> [MergedContact] {
+        let realm = getRealm()
+        let result = Array(realm.objects(MergedContact.self).filter(Self.searchContactInsensitivePredicate, string, string))
```

Comment:
Can we use `.toArray()` here?

Reply:
I tried before, and it did not work. Here is what I got : 
> Value of type 'Results<MergedContact>' has no member 'toArray'

Reply:
I actualy rewrote this part to emulate a `LIMIT` statement so we keep a manageable memory usage.